### ISSUE TITLE
Last Update (?) borked Sutro integration for me

### ISSUE THREAD

**Issue by mgoulet65** [ydogandjiev/hass-sutro#131]
I tried deleting and re-adding the integration.

No Devices or Entities display upon adding Integration. Upon trying to "Configure" I get following:

Config flow could not be loaded: 500 Internal Server Error Server got itself in trouble

No other logs or details available.

**Comment by mgoulet65** [comment]
b'A server error occurred.  Please contact the administrator.'', reason no reason
2024-06-25 07:47:07.010 ERROR (MainThread) [custom_components.sutro] Error fetching information from https://api.mysutro.com/graphql - 500, message='Internal Server Error', url=URL('https://api.mysutro.com/graphql')
2024-06-25 07:47:07.032 ERROR (MainThread) [homeassistant.components.sensor] Error adding entity None for domain sensor with platform sutro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 600, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 719, in _async_add_entity
    entity.add_to_platform_start(
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 236, in add_to_platform_start
    if self.unique_id is None or self.device_class is None:
       ^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/sensor.py", line 99, in unique_id
    return f"{self.coordinator.data['me']['device']['serialNumber']}-acidity"
              ~~~~~~~~~~~~~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable
2024-06-25 07:47:07.038 ERROR (MainThread) [homeassistant.components.sensor] Error adding entity None for domain sensor with platform sutro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 600, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 719, in _async_add_entity
    entity.add_to_platform_start(
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 236, in add_to_platform_start
    if self.unique_id is None or self.device_class is None:
       ^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/sensor.py", line 117, in unique_id
    return f"{self.coordinator.data['me']['device']['serialNumber']}-alkalinity"
              ~~~~~~~~~~~~~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable
2024-06-25 07:47:07.040 ERROR (MainThread) [homeassistant.components.sensor] Error adding entity None for domain sensor with platform sutro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 600, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 719, in _async_add_entity
    entity.add_to_platform_start(
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 236, in add_to_platform_start
    if self.unique_id is None or self.device_class is None:
       ^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/sensor.py", line 138, in unique_id
    return f"{self.coordinator.data['me']['device']['serialNumber']}-chlorine"
              ~~~~~~~~~~~~~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable
2024-06-25 07:47:07.041 ERROR (MainThread) [homeassistant.components.sensor] Error adding entity None for domain sensor with platform sutro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 600, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 719, in _async_add_entity
    entity.add_to_platform_start(
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 236, in add_to_platform_start
    if self.unique_id is None or self.device_class is None:
       ^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/sensor.py", line 177, in unique_id
    return f"{self.coordinator.data['me']['device']['serialNumber']}-temperature"
              ~~~~~~~~~~~~~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable
2024-06-25 07:47:07.042 ERROR (MainThread) [homeassistant.components.sensor] Error adding entity None for domain sensor with platform sutro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 600, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 719, in _async_add_entity
    entity.add_to_platform_start(
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 236, in add_to_platform_start
    if self.unique_id is None or self.device_class is None:
       ^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/sensor.py", line 195, in unique_id
    return f"{self.coordinator.data['me']['device']['serialNumber']}-battery"
              ~~~~~~~~~~~~~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable
2024-06-25 07:47:07.044 ERROR (MainThread) [homeassistant.components.sensor] Error adding entity None for domain sensor with platform sutro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 600, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 719, in _async_add_entity
    entity.add_to_platform_start(
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 236, in add_to_platform_start
    if self.unique_id is None or self.device_class is None:
       ^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/sensor.py", line 213, in unique_id
    return f"{self.coordinator.data['me']['device']['serialNumber']}-charges"
              ~~~~~~~~~~~~~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable
2024-06-25 07:47:07.045 ERROR (MainThread) [homeassistant.components.sensor] Error adding entity None for domain sensor with platform sutro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 600, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 719, in _async_add_entity
    entity.add_to_platform_start(
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 236, in add_to_platform_start
    if self.unique_id is None or self.device_class is None:
       ^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/sensor.py", line 159, in unique_id
    return f"{self.coordinator.data['me']['device']['serialNumber']}-bromine"
              ~~~~~~~~~~~~~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable
2024-06-25 07:47:07.046 ERROR (MainThread) [homeassistant.components.sensor] Error adding entity None for domain sensor with platform sutro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 600, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 719, in _async_add_entity
    entity.add_to_platform_start(
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 236, in add_to_platform_start
    if self.unique_id is None or self.device_class is None:
       ^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/sensor.py", line 232, in unique_id
    return f"{self.coordinator.data['me']['device']['serialNumber']}-health"
              ~~~~~~~~~~~~~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable
2024-06-25 07:47:07.047 ERROR (MainThread) [homeassistant.components.sensor] Error adding entity None for domain sensor with platform sutro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 600, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 719, in _async_add_entity
    entity.add_to_platform_start(
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 236, in add_to_platform_start
    if self.unique_id is None or self.device_class is None:
       ^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/sensor.py", line 251, in unique_id
    return f"{self.coordinator.data['me']['device']['serialNumber']}-charger-status"
              ~~~~~~~~~~~~~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable
2024-06-25 07:47:07.048 ERROR (MainThread) [homeassistant.components.sensor] Error adding entity None for domain sensor with platform sutro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 600, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 719, in _async_add_entity
    entity.add_to_platform_start(
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 236, in add_to_platform_start
    if self.unique_id is None or self.device_class is None:
       ^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/sensor.py", line 270, in unique_id
    return f"{self.coordinator.data['me']['device']['serialNumber']}-hub-ssid"
              ~~~~~~~~~~~~~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable
2024-06-25 07:47:07.050 ERROR (MainThread) [homeassistant.components.binary_sensor] Error adding entity None for domain binary_sensor with platform sutro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 600, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 743, in _async_add_entity
    if entity.unique_id is not None:
       ^^^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/binary_sensor.py", line 62, in unique_id
    device_serial = self.coordinator.data["me"]["device"]["serialNumber"]
                    ~~~~~~~~~~~~~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable
2024-06-25 07:47:07.052 ERROR (MainThread) [homeassistant.components.binary_sensor] Error adding entity None for domain binary_sensor with platform sutro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 600, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 743, in _async_add_entity
    if entity.unique_id is not None:
       ^^^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/binary_sensor.py", line 84, in unique_id
    device_serial = self.coordinator.data["me"]["device"]["serialNumber"]
                    ~~~~~~~~~~~~~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable
2024-06-25 07:47:07.053 ERROR (MainThread) [homeassistant.components.binary_sensor] Error adding entity None for domain binary_sensor with platform sutro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 600, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 743, in _async_add_entity
    if entity.unique_id is not None:
       ^^^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/binary_sensor.py", line 154, in unique_id
    device_serial = self.coordinator.data["me"]["device"]["serialNumber"]
                    ~~~~~~~~~~~~~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable
2024-06-25 07:47:07.054 ERROR (MainThread) [homeassistant.components.binary_sensor] Error adding entity None for domain binary_sensor with platform sutro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 600, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 743, in _async_add_entity
    if entity.unique_id is not None:
       ^^^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/binary_sensor.py", line 107, in unique_id
    device_serial = self.coordinator.data["me"]["device"]["serialNumber"]
                    ~~~~~~~~~~~~~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable
2024-06-25 07:47:07.055 ERROR (MainThread) [homeassistant.components.binary_sensor] Error adding entity None for domain binary_sensor with platform sutro
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 600, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 743, in _async_add_entity
    if entity.unique_id is not None:
       ^^^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/binary_sensor.py", line 130, in unique_id
    device_serial = self.coordinator.data["me"]["device"]["serialNumber"]
                    ~~~~~~~~~~~~~~~~~~~~~^^^^^^
TypeError: 'NoneType' object is not subscriptable
2024-06-25 07:47:12.867 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 452, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_app.py", line 543, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_middlewares.py", line 114, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 92, in security_filter_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 83, in forwarded_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 26, in request_context_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 85, in ban_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 242, in auth_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/headers.py", line 32, in headers_middleware
    response = await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/http.py", line 73, in handle
    result = await handler(request, **request.match_info)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/decorators.py", line 81, in with_admin
    return await func(self, request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 262, in post
    return await super().post(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 70, in wrapper
    return await method(view, request, data, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 77, in post
    return await self._post_impl(request, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 84, in _post_impl
    result = await self._flow_mgr.async_init(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 343, in async_init
    result = await self._async_handle_step(flow, flow.init_step, data)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 517, in _async_handle_step
    result: _FlowResultT = await getattr(flow, method)(user_input)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/config_flow.py", line 99, in async_step_init
    return await self.async_step_user()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/config_flow.py", line 111, in async_step_user
    vol.Required(x, default=self.options.get(x, True)): bool
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/voluptuous/schema_builder.py", line 1152, in __init__
    super(Required, self).__init__(schema, msg=msg,
  File "/usr/local/lib/python3.12/site-packages/voluptuous/schema_builder.py", line 982, in __init__
    self._schema = Schema(schema_)
                   ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/voluptuous/schema_builder.py", line 207, in __init__
    self._compiled = self._compile(schema)
                     ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/voluptuous/schema_builder.py", line 302, in _compile
    raise er.SchemaError('unsupported schema data type %r' %
voluptuous.error.SchemaError: unsupported schema data type 'Platform'
2024-06-25 07:49:09.256 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 452, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_app.py", line 543, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_middlewares.py", line 114, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 92, in security_filter_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 83, in forwarded_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 26, in request_context_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 85, in ban_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 242, in auth_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/headers.py", line 32, in headers_middleware
    response = await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/http.py", line 73, in handle
    result = await handler(request, **request.match_info)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/decorators.py", line 81, in with_admin
    return await func(self, request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 262, in post
    return await super().post(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 70, in wrapper
    return await method(view, request, data, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 77, in post
    return await self._post_impl(request, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 84, in _post_impl
    result = await self._flow_mgr.async_init(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 343, in async_init
    result = await self._async_handle_step(flow, flow.init_step, data)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 517, in _async_handle_step
    result: _FlowResultT = await getattr(flow, method)(user_input)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/config_flow.py", line 99, in async_step_init
    return await self.async_step_user()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/sutro/config_flow.py", line 111, in async_step_user
    vol.Required(x, default=self.options.get(x, True)): bool
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/voluptuous/schema_builder.py", line 1152, in __init__
    super(Required, self).__init__(schema, msg=msg,
  File "/usr/local/lib/python3.12/site-packages/voluptuous/schema_builder.py", line 982, in __init__
    self._schema = Schema(schema_)
                   ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/voluptuous/schema_builder.py", line 207, in __init__
    self._compiled = self._compile(schema)
                     ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/voluptuous/schema_builder.py", line 302, in _compile
    raise er.SchemaError('unsupported schema data type %r' %
voluptuous.error.SchemaError: unsupported schema data type 'Platform'

**Comment by EtothaJ** [comment]
I have the same issue and getting the same.  Was working perfectly until yesterday. :(

**Comment by joberdick** [comment]
Broken here too. I assume sutro made a change. I have not made any changes.

**Comment by Intecpsp** [comment]
I have this issue as well, I suspected it was a Sutro change

**Comment by joberdick** [comment]
It appears the sutro graph api is down. I opened a support ticket for some insight.

**Comment by jeffrapp** [comment]
The Sutro graph api is working for me.  I don't know if the response object has changed, though. 

**Request**
```
curl --request POST \
  --url https://api.mysutro.com/graphql \
  --header 'Authorization: Bearer {TOKEN}' \
  --header 'Connection: Keep-Alive' \
  --header 'Content-Type: application/json' \
  --data '{
	"query": "{\nme {\nid\nfirstName\ndevice {\nbatteryLevel\nserialNumber\ntemperature\ncartridgeCharges\nhealth\ncoreStatus\nlidOpen\nonline\nshouldTakeReadings\nlastMessage\ncurrentFirmwareVersion\n}\nhub {\nonline\nchargerStatus\nssid\nlastMessage\n}\npool {\nlatestReading {\nalkalinity\nbromine\nchlorine\nph\nreadingTime\n}\n}\n}\n}"
}'
```


**Response**
```
{
	"data": {
		"me": {
			"device": {
				"batteryLevel": 96,
				"cartridgeCharges": 45,
				"coreStatus": true,
				"currentFirmwareVersion": "2.1.7",
				"health": "GOOD",
				"lastMessage": "2024-06-25T15:08:27Z",
				"lidOpen": false,
				"online": true,
				"serialNumber": "1234:xxxx:xxxx:1234",
				"shouldTakeReadings": true,
				"temperature": 86.5
			},
			"firstName": "{firstName}",
			"hub": {
				"chargerStatus": "CHARGED",
				"lastMessage": "2024-06-25T15:59:07Z",
				"online": true,
				"ssid": "{ssid}"
			},
			"id": "1234",
			"pool": {
				"latestReading": {
					"alkalinity": 85.5,
					"bromine": null,
					"chlorine": 3.51,
					"ph": 7.28,
					"readingTime": "2024-06-25T15:07:53Z"
				}
			}
		}
	}
}
```

**Comment by jeffrapp** [comment]
Something must have changed with how Sutro handles GraphQL queries. I got it fixed locally, and I'll try to get a pull request built out here later.

The gist of the changes are in `api.py`, at 157, I made these updates:
```
payload = {
            "query": query,
        }
        headers = {
            "Content-Type": "application/json",
            "Authorization": f"Bearer {self._token}",
        }
        response = await self.api_wrapper("post", SUTRO_GRAPHSQL_URL, json.dumps(payload), headers)
```

**Comment by jeffrapp** [comment]
Ok, PR is in for review. 

In the meantime, you can add my version as a custom repository to HACS if you want to get your data logging again:
https://github.com/jeffrapp/hass-sutro

**Comment by EtothaJ** [comment]
> Something must have changed with how Sutro handles GraphQL queries. I got it fixed locally, and I'll try to get a pull request built out here later.
> 
> The gist of the changes are in `api.py`, at 157, I made these updates:
> 
> ```
> payload = {
>             "query": query,
>         }
>         headers = {
>             "Content-Type": "application/json",
>             "Authorization": f"Bearer {self._token}",
>         }
>         response = await self.api_wrapper("post", SUTRO_GRAPHSQL_URL, json.dumps(payload), headers)
> ```

Implemented your changed and worked like a charm.  THANK YOU!

**Comment by nolsen311** [comment]
confirmed fixed in 0.0.9

**Comment by mgoulet65** [comment]
Confirmed fixed for me as well.

**Comment by Intecpsp** [comment]
Yup, same here!

**Comment by ydogandjiev** [comment]
Something changed on the Sutro side. Resolved by #133. Thank you for the fix, @jeffrapp!
